### ISSUE TITLE
crafted markdown+front matter hangs sitegen under caddy v0.7.4

### ISSUE THREAD

**Issue by joshix** [joshix/caddybox#10]
(prelim)
<strike>TODO:joshix - Test and describe repeatably.</strike>
TODO: debugger

This is against caddybox with latest mholt/caddy@36fa6e857bd22a59017ad703414c42e19a5cc1c7 at `/bin/caddy`, binary built on /n/core3a/rkt/goer2, container built on /n/maxton/stevedore

Uncertain if bug in my build/pkging, or in upstream caddy

`./Caddyfile`:

```
0.0.0.0:2015
  markdown / {
    ext .md
    sitegen
  }
```

```
$ docker run -d -p 80:2015 -v `pwd`:/var/www:rw joshix/caddy
$ docker stats murfreesboro_donut
CONTAINER           CPU %               MEM USAGE/LIMIT     MEM %               NET I/O
local.maxton.9      147.94%             1.479 MB/650.5 MB   0.23%               5.956 kB/738 B
```

Recent commits around sitegen, markdown, in https://github.com/mholt/caddy/commit/36fa6e857bd22a59017ad703414c42e19a5cc1c7 & several parents

**Comment by joshix** [comment]
actually seems to be a bad markdown file. Can repeat outside of appC/docker container.

**Comment by joshix** [comment]
`Caddy`, no container. Markdown + sitegen + "front matter" in `bad.md` results in 100% cpu tight loop hang by caddy. The `generated_site` directory is not created, requests are not answered.

```
$ ls
Caddyfile    bad.md
$ cat Caddyfile
0.0.0.0:2015
markdown / {
  ext .md
  sitegen
}
$ cat bad.md
{
 "title" : "CHS Maxton"
}

# *Maxton* (`maxton.local`)

Hello. My name comes from McLean's nickname for his Ideal-X, the name of
his home town, Maxton. I am the second of the *docker*(1) "container ships"
in the labs.

![the Ideal X](img/IdealX.640px.png)

* [Caddy](http://caddyserver.com)

$ ../go/src/github.com/mholt/caddy/caddy
0.0.0.0:2015
$ ps aux
j     12780  100  1.2  11552  8068 pts/5    Sl+  22:39   0:32 ../go/src/github.com/mholt/caddy/caddy
$ ls
Caddyfile    bad.md     #Nb there is no `generated_site.
$ curl http://localhost:2015/bad.md
curl: (7) Failed to connect to localhost port 2015: Connection refused
```

**Comment by joshix** [comment]
- Repeated on darwin amd64 go1.4.2, exactly as above.
- Static or dynamic linking.
- Repeatable with caddyserver.com/download binary.

**Comment by joshix** [comment]
OSX 10.10.4: 188% cpu and no `generated_site` dir created

```
$ curl -o caddy_darwin_amd64_custom.tar.gz 'https://caddyserver.com/download/build?os=darwin&arch=amd64&features='
$ tar xzf caddy_darwin_amd64_custom.tar.gz 
$ sha1sum caddy
c248994841e3dede96d11da40a6af03529b7ce52    caddy
$ cd ~/badmd
$ ../caddydist/caddy &
0.0.0.0:2015
Warning: File descriptor limit 256 is too low for production sites.
$ ps auwwx |grep caddy |grep -v grep 
j            17938 188.6  0.1 145170484   5044 s003  R    12:24AM   2:00.72 ../caddydist/caddy
```

**Comment by joshix** [comment]
upstream bug mholt/caddy#211

**Comment by joshix** [comment]
Unintentionally closed.

**Comment by joshix** [comment]
Closed with upstream mholt/caddy#211. Tested and works on darwin amd64.
